modules/tensor_mechanics/include/materials/FiniteStrainUObasedCP.h

```diff
@@ -0,0 +1,295 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef FINITESTRAINUOBASEDCP_H
+#define FINITESTRAINUOBASEDCP_H
+
+#include "ComputeStressBase.h"
+
+#include "CrystalPlasticitySlipRate.h"
+#include "CrystalPlasticitySlipResistance.h"
+#include "CrystalPlasticityStateVariable.h"
+#include "CrystalPlasticityStateVariableEvolutionRateComponent.h"
+
+class FiniteStrainUObasedCP;
+
+template<>
+InputParameters validParams<FiniteStrainUObasedCP>();
+
+/**
+ * FiniteStrainUObasedCP uses the multiplicative decomposition of deformation gradient
+ * and solves the PK2 stress residual equation at the intermediate configuration to evolve the material state.
+ * The internal variables are updated using an interative predictor-corrector algorithm.
+ * Backward Euler integration rule is used for the rate equations.
+ *
+ * Involves 4 different types of user objects that calculates:
+ * State variables - update state variable (derive from CrystalPlasticityStateVariable)
+ * State variable evolution compoment - individual component of state variable incremental rate (derive from CrystalPlasticityStateVariableEvolutionRateComponent)
+ * Slip resistance - calcuate slip resistances (derive from CrystalPlasticitySlipResistances)
+ * Slip rates - calcuate flow direction and slip rates (derive from CrystalPlasticitySlipRates)
+ */
+class FiniteStrainUObasedCP : public ComputeStressBase
+{
+public:
+  FiniteStrainUObasedCP(const InputParameters & parameters);
+
+protected:
+  /**
+   * updates the stress at a quadrature point.
+   */
+  virtual void computeQpStress();
+
+  /**
+   * updates the elasticity tensor at a quadrature point.
+   * Presently void.
+   */
+  virtual void computeQpElasticityTensor();
```

Comment:
Is this function required since elasticity tensor is computed in a different material class?